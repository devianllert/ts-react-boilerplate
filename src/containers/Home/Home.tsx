import React, { ReactElement } from 'react';
import { Helmet } from 'react-helmet';

import MainLayout from '../../layout/MainLayout';

const Home = (): ReactElement => (
  <MainLayout>
    <Helmet>
      <title>Home</title>
    </Helmet>

    <h1>React Typescript Boilerplate</h1>
  </MainLayout>
);

export default Home;
