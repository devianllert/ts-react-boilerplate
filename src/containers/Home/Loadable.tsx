import loadable, { ImportComponent } from '../../utils/loadable';

const LoadableHome = loadable((): ImportComponent => import(/* webpackChunkName: "home" */ './Home'));

export default LoadableHome;
