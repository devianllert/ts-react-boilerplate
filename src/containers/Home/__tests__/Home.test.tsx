import React from 'react';
import { cleanup, render } from '@testing-library/react';

import Home from '../Home';

import '@testing-library/jest-dom/extend-expect';

describe('<Home />', (): void => {
  afterEach(cleanup);

  it('should render and match the snapshot', (): void => {
    const { container } = render(<Home />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with initial state', (): void => {
    const { getByText } = render(<Home />);

    expect(getByText(/React Typescript Boilerplate/)).toBeDefined();
  });
});
