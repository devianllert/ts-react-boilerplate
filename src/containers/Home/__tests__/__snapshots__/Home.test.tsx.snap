// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render and match the snapshot 1`] = `
<div
  class="container"
>
  <h1>
    React Typescript Boilerplate
  </h1>
</div>
`;
